internal/cli/atlas/streams/instance/create.go

```diff
@@ -0,0 +1,118 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package instance
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20230201004/admin"
+)
+
+type CreateOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	cli.InputOpts
+	name     string
+	provider string
+	region   string
+	store    store.StreamsCreator
+}
+
+const (
+	createTemplate = "Atlas Streams Processor Instance '{{.Name}}' successfully created.\n"
+)
+
+func (opts *CreateOpts) Run() error {
+	streamProcessor := atlasv2.NewStreamsTenant()
+	streamProcessor.Name = &opts.name
+	streamProcessor.GroupId = &opts.ProjectID
+	streamProcessor.DataProcessRegion = atlasv2.NewStreamsDataProcessRegion(opts.provider, opts.region)
+
+	r, err := opts.store.CreateStream(opts.ProjectID, streamProcessor)
+
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *CreateOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+// CreateBuilder
+// atlas streams instance create [name]
+// --provider AWS
+// --region VIRGINIA_USA.
+func CreateBuilder() *cobra.Command {
+	opts := &CreateOpts{}
+	cmd := &cobra.Command{
+		Use:   "create [name]...",
+		Short: "Create an Atlas Stream Processor Instance for your project",
+		Long:  `To get started quickly, specify a name, a cloud provider, and a region to configure an Atlas Streams processor instance.` + fmt.Sprintf(usage.RequiredRole, "Project Owner"),
+		Example: fmt.Sprintf(`  # Deploy an Atlas Streams provider instance called myProcessor for the project with the ID 5e2211c17a3e5a48f5497de3:
+  %[1]s streams create myProcessor --projectId 5e2211c17a3e5a48f5497de3 --provider AWS --region VIRGINIA_USA`, cli.ExampleAtlasEntryPoint()),
+		Args: require.MaximumNArgs(1),
+		Annotations: map[string]string{
+			"nameDesc": "Name of the Atlas Streams processor instance. The instance name cannot be changed after the processor is created. The name can contain ASCII letters, numbers, and hyphens.",
+			"output":   createTemplate,
+		},
+		ValidArgs: []string{"name"},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("the Atlas Streams Processor instance name is missing")
+			}
+
+			if len(args) != 0 {
+				opts.name = args[0]
+			}
```

Comment:
is it mandatory or not? why not use the validation to make it mandatory and document acoordingly -> `[name]` should be `<name>` if mandatory

Reply:
Was resolved in [1cb5065](https://github.com/mongodb/mongodb-atlas-cli/pull/2138/commits/1cb5065fed04f4401642cd750a2e1d0c6e4622cb)